app/assets/javascripts/controllers/bootstrap_tree_controller.js

```diff
@@ -0,0 +1,58 @@
+ManageIQ.angular.app.controller('bootstrapTreeController', ['$http', '$scope', 'bootstrapTreeSubscriptionService', 'initialTreeData', 'railsControllerName', function($http, $scope, bootstrapTreeSubscriptionService, initialTreeData, railsControllerName) {
+  var init = function() {
+    updateTree(initialTreeData);
+    $scope.rootNode = $('#bootstrap-tree').treeview('getNodes')[0];
+    $('#bootstrap-tree').treeview('selectNode', $scope.rootNode);
+
+    bootstrapTreeSubscriptionService.subscribeToTreeUpdates(updateTree);
+    bootstrapTreeSubscriptionService.subscribeToCancelClicks(selectRootNode);
+    bootstrapTreeSubscriptionService.subscribeToDeselectTreeNodes(unselectAllNodes);
+    bootstrapTreeSubscriptionService.subscribeToSingleItemSelected(selectSingleNode);
+  };
+
+  var updateTree = function(data) {
+    $('#bootstrap-tree').treeview({
+      collapseIcon: 'fa fa-angle-down',
+      data: data,
+      expandIcon: 'fa fa-angle-right',
+      nodeIcon: 'fa fa-folder',
+      showBorder: false,
+      onNodeSelected: nodeSelectedCallback
+    });
+  };
+
+  var nodeSelectedCallback = function(_event, data) {
```

Comment:
`onNodeSelected` and the corresponding `nodeSelectedCallback` shouldn't be doing any requests by themselves, it's purely a javascript thing that has to do with the trees. So, anything that happens when you click on a node should just trigger.

I just attempted to prove by setting a scope variable to a random number, added an input to the html, and when I clicked around on the tree the input changed.